src/java/org/apache/cassandra/config/Converters.java

```diff
@@ -69,44 +69,47 @@
      */
     SECONDS_CUSTOM_DURATION(String.class, DurationSpec.IntSecondsBound.class,
                             DurationSpec.IntSecondsBound::inSecondsString,
-                            o -> Long.toString(o.toSeconds())),
+                            o -> o == null ? null : Long.toString(o.toSeconds())),
     /**
      * This converter is used to support backward compatibility for parameters where in the past -1 was used as a value
      * Example:  index_summary_resize_interval_in_minutes = -1 and  index_summary_resize_interval = null are equal.
      */
     MINUTES_CUSTOM_DURATION(Integer.class, DurationSpec.IntMinutesBound.class,
-                     o -> o == -1 ? null : new DurationSpec.IntMinutesBound(o),
-                     o -> o == null ? -1 : o.toMinutes()),
+                            o -> o == -1 ? null : new DurationSpec.IntMinutesBound(o),
+                            o -> o == null ? -1 : o.toMinutes()),
     MEBIBYTES_DATA_STORAGE_LONG(Long.class, DataStorageSpec.LongMebibytesBound.class,
                                 DataStorageSpec.LongMebibytesBound::new,
-                                DataStorageSpec.LongMebibytesBound::toMebibytes),
+                                o -> o == null ? null : o.toMebibytes()),
     MEBIBYTES_DATA_STORAGE_INT(Integer.class, DataStorageSpec.IntMebibytesBound.class,
                                DataStorageSpec.IntMebibytesBound::new,
-                               DataStorageSpec.IntMebibytesBound::toMebibytes),
+                               o -> o == null ? null : o.toMebibytes()),
+    NEGATIVE_MEBIBYTES_DATA_STORAGE_INT(Integer.class, DataStorageSpec.IntMebibytesBound.class,
+                              o -> o < 0 ? null : new DataStorageSpec.IntMebibytesBound(o),
+                              o -> o == null ? -1 : o.toMebibytes()),
```

Comment:
Trivial nit that can be addressed on commit: the parameters are misaligned after renaming the converter.